docs/docs/globalConfig.md
# 全局配置

#### 示例 1

<vuep template="#simple"></vuep>

<script v-pre type="text/x-template" id="simple">
<template>
    <e-line :data="data" style="width: 700px; height: 400px;background: #084a81"></e-line>
</template>

<script>
  export default {
		created () {
			this.$xEchart.setChartConfig({
				FONT_S: 16,
				THEME_COLOR: 'dark'
			});
		},
    data () {
      return {
        data: {
			xAxis: ['2014年', '2015年', '2016年', '2017年', '2018年', '2019年'],
			series: [
				{
						name: '均值',
						type: 'bar',
						data: [2, 1, 2, 4, 5, 3]
				},
				{
						name: '2018年',
						data: [2.5, 1.5, 3, 4, 5.5, 3.5]
				},
				{
						name: '2019年',
						data: [1.5, 0.5, 1, 4, 4.8, 2.5]
				}
			]
        }
      }
    }
  }
</script>
</script>

#### 示例 2

<vuep template="#simple_1"></vuep>

<script v-pre type="text/x-template" id="simple_1">
<template>
    <e-line 
        :data="data" 
        :config="{
            color: ['#d8cf3a', '#20cb44'],
            showFillArea: true
        }"
        style="width: 700px; height: 600px;"
    ></e-line>
</template>

<script>
  export default {
		created () {
			this.$xEchart.setChartConfig({
				FONT_S: 16,
				THEME_COLOR: 'light'
			});
		},
    data () {
      return {
        data: {
            xAxis: ['2014年', '2015年', '2016年', '2017年', '2018年', '2019年'],
            series: [
                {
                    name: '2018年',
                    data: [20, 33, 28, 36, 28, 35]
                },
                {
                    name: '2019年',
                    data: [28, 36, 28, 30, 22, 33]
                }
            ]
        }
      }
    }
  }
</script>
</script>

#### setChartConfig() 配置项

| 配置项       | 简介                                                                                     | 类型    | 备注                                                                            |
| ----------- | -------------------------------------------------------------------------------------- - | ------- | ------------------------------- |
| THEME_COLOR  | 图表主题色系  | string  | 默认值"light", "light"使用白底配色 "dark"使用深色底配色
| FONT_FAMILY  | 字体样式    |  string  | 默认值 'sans-serif'， 添加自定义字体需要先在样式中引入
| FONT_S       | 常规字体大小 | number  | 默认值 16 |
| FONT_COLOR   | 字体颜色  | string  | 不设置时，字体颜色跟随THEME_COLOR变动，分别取FONT_COLOR_DARK或FONT_COLOR_LIGHT的值 |
| FONT_COLOR_DARK   | 深色背景下字体颜色  | string  | 默认值 '#fff' |
| FONT_COLOR_LIGHT   | 白色背景下字体颜色  | string  | 默认值 '#666' |
| BORDER_COLOR | 轴线颜色  | string  | 不设置时，轴线颜色跟随THEME_COLOR变动，分别取BORDER_COLOR_DARK或BORDER_COLOR_LIGHT的值  |
| BORDER_COLOR_DARK   | 深色背景下字体颜色  | string  | 默认值 '#0169a2' |
| BORDER_COLOR_LIGHT   | 白色背景下字体颜色  | string  | 默认值 '#c8c8c8' |
| COLOR   | 图表配色  | array  | 不设置时，配色跟随THEME_COLOR变动，分别取COLOR_DARK或COLOR_LIGHT的值 |
| COLOR_DARK   | 深色背景下图表配色  | array  | 默认值见颜色说明 |
| COLOR_LIGHT   | 白色背景下图表配色  | array  | 默认值见颜色说明 |
| LEGEND_ICON  | 图例现状  | string  | 默认值 'rect'
| LEGEND_ICON_SIZE | 图例大小  | Number   | 默认值 16
| SHOW_TOOLBOX | 是否显示工具栏 | boolean | 默认值 false |
| BG_COLOR     | 保存为图片时的填充背景色   | string  | 默认值 "rgba(0,0,0,0)"   |
| IMG_NONE     | 无数据时的替代图片  | string  |
| IMG_NONE_W   | 无数据图片的宽度  | Number  | 默认值 217 |
| IMG_NONE_H   | 无数据图片的高度 | Number  | 默认值 134 |
| GRID         | 配置直角坐标系边距（作用于折线图、柱状图、堆叠图、热力图）<br/>参照 echarts 的 grid 配置 | Object  | 默认值 <br/>{containLabel: true,top: 50,left: '5%',right: '5%',bottom: '5%'} |
| EVENTS       | echarts添加监听的事件列表  | array   | 默认值 ['click','dblclick']
| TOOLTIP_BG_COLOR | 提示框背景色  | string   | 默认值 '#fff'
| TOOLTIP_BORDER_COLOR | 提示框边框颜色  | string   | 默认值 '#c8c8c8'
| TOOLTIP_FONT_COLOR | 提示框文本颜色  | string   | 默认值 '#666'

> 备注： <br/>setChartConfig 仅支持全局配置， 设置后即公用。<br/>如无特殊要求，不同底色下一般只需要配置THEME_COLOR设置主题色系。 <br/> 因为 echarts 全局字体样式本身无法覆盖所有字体， 因此添加自定义配置时存在字体样式不一致的情况， 需要手动调整。


docs/docs/en/globalConfig.md
# Global Configuration

#### example 1

<vuep template="#simple"></vuep>

<script v-pre type="text/x-template" id="simple">
<template>
    <e-line :data="data" style="width: 700px; height: 400px;background: #084a81"></e-line>
</template>

<script>
  export default {
		created () {
			this.$xEchart.setChartConfig({
				FONT_S: 16,
				THEME_COLOR: 'dark'
			});
		},
    data () {
      return {
        data: {
			xAxis: ['2014', '2015', '2016', '2017', '2018', '2019'],
			series: [
				{
						name: 'MeanValue',
						type: 'bar',
						data: [2, 1, 2, 4, 5, 3]
				},
				{
						name: '2018',
						data: [2.5, 1.5, 3, 4, 5.5, 3.5]
				},
				{
						name: '2019',
						data: [1.5, 0.5, 1, 4, 4.8, 2.5]
				}
			]
        }
      }
    }
  }
</script>
</script>

#### example 2

<vuep template="#simple_1"></vuep>

<script v-pre type="text/x-template" id="simple_1">
<template>
    <e-line 
        :data="data" 
        :config="{
            color: ['#d8cf3a', '#20cb44'],
            showFillArea: true
        }"
        style="width: 700px; height: 600px;"
    ></e-line>
</template>

<script>
  export default {
		created () {
			this.$xEchart.setChartConfig({
				FONT_S: 16,
				THEME_COLOR: 'light'
			});
		},
    data () {
      return {
        data: {
            xAxis: ['2014', '2015', '2016', '2017', '2018', '2019'],
            series: [
                {
                    name: '2018',
                    data: [20, 33, 28, 36, 28, 35]
                },
                {
                    name: '2019',
                    data: [28, 36, 28, 30, 22, 33]
                }
            ]
        }
      }
    }
  }
</script>
</script>

#### setChartConfig() Configuration Options


| Configuration Option | Description | Type | Remarks |
| -------------------- | ----------- | ---- | ------- |
| THEME_COLOR | Chart theme color scheme | string | Default value is "light". "light" uses a white background color scheme, "dark" uses a dark background color scheme. |
| FONT_FAMILY | Font style | string | Default value is 'sans-serif'. Adding a custom font requires prior importing in the styles. |
| FONT_SIZE | Normal font size | number | Default value is 16. |
| FONT_COLOR | Font color | string | If not set, the font color will change according to the THEME_COLOR, taking the values of FONT_COLOR_DARK or FONT_COLOR_LIGHT. |
| FONT_COLOR_DARK | Font color in dark background | string | Default value is '#fff'. |
| FONT_COLOR_LIGHT | Font color in light background | string | Default value is '#666'. |
| BORDER_COLOR | Axis line color | string | If not set, the axis line color will change according to the THEME_COLOR, taking the values of BORDER_COLOR_DARK or BORDER_COLOR_LIGHT. |
| BORDER_COLOR_DARK | Axis line color in dark background | string | Default value is '#0169a2'. |
| BORDER_COLOR_LIGHT | Axis line color in light background | string | Default value is '#c8c8c8'. |
| COLOR | Chart color scheme | array | If not set, the color scheme will change according to the THEME_COLOR, taking the values of COLOR_DARK or COLOR_LIGHT. |
| COLOR_DARK | Chart color scheme in dark background | array | Default value: see color description. |
| COLOR_LIGHT | Chart color scheme in light background | array | Default value: see color description. |
| LEGEND_ICON | Legend icon type | string | Default value is 'rect'. |
| LEGEND_ICON_SIZE | Legend icon size | number | Default value is 16. |
| SHOW_TOOLBOX | Whether to display the toolbox | boolean | Default value is false. |
| BG_COLOR | Background color when saving as an image | string | Default value is "rgba(0,0,0,0)". |
| IMG_NONE | Replacement image when there is no data | string | |
| IMG_NONE_W | Width of the image when there is no data | number | Default value is 217. |
| IMG_NONE_H | Height of the image when there is no data | number | Default value is 134. |
| GRID | Configuration for grid margins<br/>(applies to line charts, bar charts, stack charts, and heat maps) | Object | Default value: see color description. |
| EVENTS | Echarts events to listen to | array | Default value is ['click', 'dblclick']. |
| TOOLTIP_BG_COLOR | Tooltip background color | string | Default value is '#fff'. |
| TOOLTIP_BORDER_COLOR | Tooltip border color | string | Default value is '#c8c8c8'. |
| TOOLTIP_FONT_COLOR | Tooltip text color | string | Default value is '#666'. |

> remark： <br/>setChartConfig 仅支持全局配置， 设置后即公用。<br/>如无特殊要求，不同底色下一般只需要配置THEME_COLOR设置主题色系。 <br/> 因为 echarts 全局字体style本身无法覆盖所有字体， 因此添加自定义配置时存在字体style不一致的情况， 需要手动调整。


src/components/src/config/config.js
/** @format */
// import noDataPng from '../images/nodata.png';

export default {
    //主题颜色
    THEME_COLOR: 'light',
    //echarts事件
    EVENTS: [
        'click',
        'dblclick',
        /* 'mouseover',
        'mouseout',
        'mousemove',
        'mousedown',
        'mouseup' */
    ],
    //字体
    FONT_FAMILY: 'sans-serif',
    FONT_S: 16,
    FONT_M: 20,
    FONT_L: 28,
    //字体颜色
    FONT_COLOR: '',
    FONT_COLOR_DARK: '#fff',
    FONT_COLOR_LIGHT: '#666',
    //轴线颜色
    BORDER_COLOR: '',
    BORDER_COLOR_DARK: '#0169a2',
    BORDER_COLOR_LIGHT: '#c8c8c8',
    //颜色
    COLOR: null,
    COLOR_DARK: [
        '#4B97FF', '#7FFFB7', '#FDDD60', '#FF6E76', '#58D9F9', '#05C091', '#FF8A45', '#8D48E3', '#DD79FF'
    ],
    COLOR_LIGHT: [
        '#5470C6', '#91CC75', '#FACB59', '#EE6666', '#73C0DE', '#3BA272', '#FD8C56', '#9A60B4', '#EA7CCC'
    ],
    //水质Ⅰ~ 劣Ⅴ类颜色
    waterGradesColor: [
        '#44c5fd',
        '#51a5fd',
        '#73bb31',
        '#eebd15',
        '#f88e17',
        '#ee3b5b'
    ],
    //合并1，2类的水质级别颜色
    merge1n2WaterGradesColor: [
        '#44c5fd',
        '#73bb31',
        '#eebd15',
        '#f88e17',
        '#ee3b5b'
    ],

    //水质Ⅰ~ 劣Ⅴ类
    waterGradesLabel: ['Ⅰ类', 'Ⅱ类', 'Ⅲ类', 'Ⅳ类', 'Ⅴ类', '劣Ⅴ类'],

    //富营养状态等级颜色
    eutrophicationColor: [
        '#44c5fd',
        '#73bb31',
        '#eebd15',
        '#f88e17',
        '#ee3b5b'
    ],

    //空气质量 优 ~ 严重污染 颜色
    airGradesColor: [
        '#24bd5d',
        '#d8bc37',
        '#f87c12',
        '#f60000',
        '#94004b',
        '#6f001f'
    ],

    //是否显示工具栏
    SHOW_TOOLBOX: false,

    //保存为图片时的填充背景色
    BG_COLOR: 'rgba(0,0,0,0)',
    // TEXT_NONE: '暂时没有数据',
    // IMG_NONE: '',
    // IMG_NONE_W: 202,
    // IMG_NONE_H: 154,

    //边距
    GRID: {
        containLabel: true,
        top: '50',
        left: '5%',
        right: '5%',
        bottom: '5%'
    },

    LEGEND_ICON: 'rect',//图例图标
    LEGEND_ICON_SIZE: 16,//图例大小
    TOOLTIP_BG_COLOR: '#fff',//提示框背景色
    TOOLTIP_BORDER_COLOR: '#c8c8c8',//提示框边框颜色
    TOOLTIP_FONT_COLOR: '#666'//提示框文本颜色
};
